tmt/steps/provision/bootc.py

```diff
@@ -75,6 +75,12 @@ def remove(self) -> None:
             logger=self._logger,
             env=PODMAN_ENV if self._rootless else None)
 
+        # plan with bootc provision should build its own qcow2 before test run
+        # test qcow2 image should be removed when plan is done
+        tmt.utils.Command(
+            "rm", "-vf", f"{DEFAULT_TMP_PATH}/images/disk.qcow2"
```

Comment:
1. `unlink` should work as well: `(DEFAULT_TMP_PATH / 'images/disk.qcow2').unlink(missing_ok=True)`
2. but what caught my eye: is this a single path shared by all plans and all `bootc` plugin invocations? That seems like a huge risk of conflict, being a shared, mutable state without proper synchronization. Can the name be changed, to reflect which run, plan, and guest it belongs to? Because this is not good, if all runs and plans all compete over the same file, we are just asking for trouble. Either its name or path should be different, perhaps the file should be moved under a workdir, something like this.

Reply:
Thanks for the comment.

The bib build `disk.qocw2` and save into `{self.workdir}/qcow2/disk.qcow2`. The `workdir` is different for each plan.

The `disk.qcow2` should be copied to DEFAULT_TMP_PATH}/images by `virtual.testcloud` when testcloud run VM. Is this correct?

What happens when defined qcow2 path https://github.com/teemtee/tmt/blob/5dc0a1f4441a7a3727fc36160302355d7d737363/tmt/steps/provision/bootc.py#L318C31-L318C62 for each plan and run it https://github.com/teemtee/tmt/blob/5dc0a1f4441a7a3727fc36160302355d7d737363/tmt/steps/provision/bootc.py#L344?

The name `disk.qcow2` is hardcoded by bib.

Reply:
`data.image = f"file://{self.workdir}/qcow2/disk.qcow2"` looks good, that's what I would expect to exist.

> The disk.qcow2 should be copied to DEFAULT_TMP_PATH}/images by virtual.testcloud when testcloud run VM. Is this correct?

I see. No, I think this spells trouble. I don't mean now, `bootc` plugin is basically a tech preview, but once more people start playing with it and someone tries more than one `provision` phase in their plan, multiple `bootc`, it may easily become a problem as `virtual` would use the same qcow2 file for all of them. Which may not necessarily be the right choice...

I think it's something left to resolve, sooner rather than later, and we might need help from testcloud owners.

Reply:
Thanks @happz! I'm not tmt expert, so anyone can help on this?